### PR TITLE
CMS-890: Adds tab for the wp pantheon cache commands.

### PR DESCRIPTION
**Pull request by rwagner00**
Closes https://getpantheon.atlassian.net/browse/CMS-890

## Summary

**[Pantheon WP-CLI Commands](https://pantheon.io/docs/guides/wp-cli/wp-cli-pantheon-commands)** - Adds documentation about commands added by the Pantheon mu-plugin to control caching on WP sites via the WP-CLI.

**Release**:
- [X] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

File: source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+cms: "WordPress"
+categories: [develop]
+tags: [wp-cli, cli]
+layout: guide
+showtoc: true
+permalink: docs/guides/wp-cli/wp-cli-pantheon-commands
+anchorid: wp-cli-pantheon-commands
+---
+
+This section provides information on how to run Pantheon specific commands on WP-CLi.
```
Comment (jspellman814) [inline source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md:14]:
```suggestion
This section provides information on how to run Pantheon specific commands on WP-CLI.
```

File: source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Prerequisites
+
+The below functionality relies on the [Pantheon Must-Use Plugin](https://github.com/pantheon-systems/pantheon-mu-plugin). 
+For installation instructions, view the Must-Use plugin README.  
+
+## Control cache with WP-CLI
+
+To enable maintenance mode while serving cached pages to visitors, use the command `pantheon cache set-maintenance-mode {MODE}`
```
Comment (jspellman814) [inline source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md:23]:
```suggestion
To enable maintenance mode while serving cached pages to visitors, use the command `wp pantheon cache set-maintenance-mode {MODE}`
```

File: source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+The Mode options are: 
+* `disabled` -- Disables maintenance mode and return to normal operation
+* `anonymous` -- Serves cached pages to visitors and bots
+* `everyone` -- Serves cached pages to all visitors _except_ administrators
+
+## More Resources
+
```
Comment (jspellman814) [inline source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md:30]:
```suggestion
- [WP-CLI on the Pantheon Platform](/guides/wp-cli)
```

**Review by jspellman814 — CHANGES_REQUESTED** [review]

**Review by jazzsequence — COMMENTED** [review]
I will come back with more feedback. There are other WP-CLI commands that this issue/doc should cover. The objective of CMS-890 was to document all the `wp pantheon` subcommands because we don't have those documented anywhere (outside of WP-CLI itself). I will find all the subcommands and list them after PBR but I wanted to drop this note here before I forgot.

**Review by jazzsequence — CHANGES_REQUESTED** [review]
Under `wp pantheon` there are two subcommands

```
usage: wp pantheon cache <command>
   or: wp pantheon session <command>
```

running `terminus wp -- <site>.<env> pantheon <subcommand>` on either of those will give you usage examples of what they expect and running `terminus wp -- <site>.<env> help pantheon <subcommand>` will give you the full description (these also exist in the code, but since the code is spread across a couple different things -- the mu-plugin and the Pantheon Native PHP Sessions plugin (I believe), there's not a single source of truth besides WPCLI itself to get those descriptions). Essentially, these docs should include the same information you can get from the commandline so as a user you know they exist and can be used (which isn't super obvious otherwise). It might also be helpful to specify where the commands come from (e.g. which things come from the mu-plugin and which things come from the Native PHP Sessions plugin).

File: source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+permalink: docs/guides/wp-cli/wp-cli-pantheon-commands
+anchorid: wp-cli-pantheon-commands
+---
+
+This section provides information on how to run Pantheon specific commands on WP-CLi.
+
+## Prerequisites
+
+The below functionality relies on the [Pantheon Must-Use Plugin](https://github.com/pantheon-systems/pantheon-mu-plugin). 
+For installation instructions, view the Must-Use plugin README.
```
[the remark this answers is not in the document]
Comment (rwagner00) [reply source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md:19]: That's fair, I'll remove that line.

**Comment by rwagner00** [comment]
@jazzsequence Hrmm, running `terminus wp -- jrw-wp-test.dev pantheon cache` produces only the set-maintenance-mode subcommand, which is the one documented. Running `terminus wp -- jrw-wp-test.dev pantheon session` produces an error that there is no such command. This is on a brand new site using the standard WP upstream.  Is there something I'm missing?

**Comment by jazzsequence** [comment]
@rwagner00 You're probably missing the [Native PHP Sessions plugin](https://github.com/pantheon-systems/wp-native-php-sessions) or you have it but it's not active. I think the other caching functions are coming from [Pantheon Advanced Page Cache](https://github.com/pantheon-systems/pantheon-advanced-page-cache).

_I'm_ missing the maintenance mode plugin, and so I'm wondering what I'm missing in my environment but IIRC (and since your site doesn't have any other Pantheon plugins installed) that should be in the mu-plugin. If you've cloned your site locally, does it have a `wp-content/mu-plugins/pantheon/cli.php` file?

**Comment by rwagner00** [comment]
@jazzsequence Neither the Native PHP nor the Advanced Page Cache plugins are included by default, per their own README files and a review of my stock site, which is why I didn't have those commands. With that in mind, I think the sections for those belong in their own tabs. Does that make sense to you?

**Comment by rwagner00** [comment]
@jazzsequence I believe I've resolved the feedback and issues, let me know if there's anything else I need to take care of!

File: source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+subtitle: Pantheon Commands for WP-CLI
+description: Special commands to interact with Pantheon via WP-CLI
+cms: "WordPress"
+categories: [develop]
+tags: [wp-cli, cli]
+layout: guide
+showtoc: true
+permalink: docs/guides/wp-cli/wp-cli-pantheon-commands
+anchorid: wp-cli-pantheon-commands
+---
```
Comment (EdwardAngert) [inline source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md:12]:
```suggestion
anchorid: wp-cli-pantheon-commands
reviewed: "2022-10-06"
---
```

**Review by EdwardAngert — CHANGES_REQUESTED** [review]
some consistency and md adjustments

File: source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+title: WP-CLI on the Pantheon Platform
+subtitle: Pantheon Commands for WP-CLI
+description: Special commands to interact with Pantheon via WP-CLI
```
Comment (EdwardAngert) [inline source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md:4]:
```suggestion
description: Special commands to interact with Pantheon via WP-CLI
contributors: [ryanwagner]
```

**Review by jazzsequence — APPROVED** [review]
Agree with @EdwardAngert's suggestions but other than that, from a technical standpoint, this looks g2g to me.

File: source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+This section provides information on how to run Pantheon specific commands on WP-CLI.
+
+## Control cache with WP-CLI
+
+### Maintenance Mode
+To enable maintenance mode while serving cached pages to visitors, use the command `wp pantheon cache set-maintenance-mode {MODE}`
+The Mode options are: 
+* `disabled` -- Disables maintenance mode and return to normal operation
+* `anonymous` -- Serves cached pages to visitors and bots
+* `everyone` -- Serves cached pages to all visitors _except_ administrators
```
[the remark this answers is not in the document]
Comment (EdwardAngert) [reply source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md:25]: @jspellman814 @rwagner00 just want to confirm this before I squashmerge this pr

1. is the command

   ```bash
   wp pantheon cache set-maintenance-mode {everyone}
   ```

   or

   ```bash
   wp pantheon cache set-maintenance-mode everyone
   ```

2. will the command ever be used in combination with another or is it always `terminus wp pantheon...`

   ```bash
   terminus wp pantheon cache set-maintenance-mode everyone
   ```

[the remark this answers is not in the document]
Comment (rwagner00) [reply source/content/guides/wp-cli/06-wp-cli-pantheon-commands.md:25]: 1) It's the latter, the brackets are just to show that it's a variable and you should replace that with your desired value.

2) Not that I'm aware of, and if it does wind up being used commonly in conjunction with another command we should just update the docs here.

**Comment by EdwardAngert** [comment]
[preview](https://pr-7753--pantheon-docs.my.pantheonfrontend.website/docs/guides/wp-cli/wp-cli-pantheon-commands)

**Review by EdwardAngert — APPROVED** [review]

**Comment by EdwardAngert** [comment]
@rwagner00 @jspellman814 before I move this to a copy review, I'm getting some errors in testing - `set-maintenance-mode everyone` worked, but the cache commands are giving me this:

```bash
Error: 'purge-all' is not a registered subcommand of 'pantheon cache'. See 'wp help pantheon cache' for available subcommands.
 [notice] Command: $SITE.dev -- wp help pantheon cache purge-all [Exit: 1]
```

for good measure, here's the help output

```bash
╰─❯ terminus wp -- $SITE.dev help pantheon cache
 [warning] This environment is in read-only Git mode. If you want to make changes to the codebase of this site (e.g. updating modules or plugins), you will need to toggle into read/write SFTP mode first.
NAME

  wp pantheon cache

SYNOPSIS

  wp pantheon cache <command>

SUBCOMMANDS

  set-maintenance-mode      Sets maintenance mode status.





GLOBAL PARAMETERS

  --path=<path>
      Path to the WordPress files.

  --url=<url>
      Pretend request came from given URL. In multisite, this argument is how
      the target site is specified.

  --ssh=[<scheme>:][<user>@]<host|container>[:<port>][<path>]
      Perform operation against a remote server over SSH (or a container using
      scheme of "docker", "docker-compose", "vagrant").

  --http=<http>
      Perform operation against a remote WordPress installation over HTTP.

  --user=<id|login|email>
      Set the WordPress user.

  --skip-plugins[=<plugins>]
      Skip loading all plugins, or a comma-separated list of plugins. Note:
      mu-plugins are still loaded.

  --skip-themes[=<themes>]
      Skip loading all themes, or a comma-separated list of themes.

  --skip-packages
      Skip loading all installed packages.

  --require=<path>
      Load PHP file before running the command (may be used more than once).

  --exec=<php-code>
      Execute PHP code before running the command (may be used more than once).

  --context=<context>
      Load WordPress in a given context.

  --[no-]color
      Whether to colorize the output.

  --debug[=<group>]
      Show all PHP errors and add verbosity to WP-CLI output. Built-in groups
      include: bootstrap, commandfactory, and help.

  --prompt[=<assoc>]
      Prompt the user to enter values for all command arguments, or a subset
      specified as comma-separated values.

  --quiet
      Suppress informational messages.

 [notice] Command: $SITE.dev -- wp help pantheon cache [Exit: 0]
```

**Comment by rwagner00** [comment]
I'll take a look.

**Comment by rwagner00** [comment]
@EdwardAngert The pantheon-advanced-page-cache plugin is inactive by default, even though it is also included in the composer file. If you activate the plugin, the commands appear. Opinions on if we should note this in the doc, or enable it by default?

**Comment by EdwardAngert** [comment]
@rwagner00 thanks! I was able to activate it and/but I'm torn about how much users want things active by default, and as far as documenting it goes, I added a few command examples in the [latest commit](https://github.com/pantheon-systems/documentation/pull/7753/commits/63757c2e6eea4424b66eeda51af1ed4a48d7f281) that I hope will help folks run them

re: `pantheon session list`
I'm not sure how to give folks instructions to delete a session (because I'm not sure how to trigger a session), so it's a little incomplete for now ([L72](https://github.com/pantheon-systems/documentation/pull/7753/files#diff-9d105b050d65ad799aa6dd0d0fd606fc1e031077e2b9959a5e793700f95d481bR72))

```bash
╰─❯ terminus wp $SITE.$ENV -- pantheon session list
+------------+---------+----------+------------+------+
| session_id | user_id | datetime | ip_address | data |
+------------+---------+----------+------------+------+
+------------+---------+----------+------------+------+
 [notice] Command: $SITE.$ENV -- wp pantheon session list [Exit: 0]
```

other than that, if it looks ready to y'all, I'll pass it on to the rest of the Docs team for a copy review

**Comment by rwagner00** [comment]
@EdwardAngert I don't actually know anything more about the PHP Session module than you do, so I'm useless there. The rest looks good to me!

**Comment by EdwardAngert** [comment]
@pantheon-systems/docs-admins when you merge, please use the **Squash and merge** option

**Comment by whitneymeredith** [comment]
LGTM

**Pull request merged**